openslides/users/views.py

```diff
@@ -478,3 +162,58 @@ def get_form_kwargs(self):
         kwargs = super().get_form_kwargs()
         kwargs['user'] = self.request.user
         return kwargs
+
+
+class UserLoginView(AjaxMixin, View):
+    """
+    Login the user via ajax.
+    """
+
+    def get_ajax_context(self, **context):
+        if self.success:
+            context.update({
+                'success': True,
+                'user_id': self.user.pk})
+        else:
+            context.update({'success': False})
+
+        return super().get_ajax_context(**context)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            data = json.loads(request.body.decode('utf-8'))
+        except ValueError:
+            self.success = False
+        else:
+            form = AuthenticationForm(request, data=data)
+            if form.is_valid():
+                self.user = form.get_user()
+                auth_login(request, self.user)
+                self.success = True
+            else:
+                self.success = False
+        return self.ajax_response()
+
+
+class UserLogoutView(AjaxMixin, View):
+    """
+    Logout the user via ajax.
+    """
+
+    def post(self, request, *args, **kwargs):
+        auth_logout(request)
+        return self.ajax_response()
+
+
+class WhoAmIView(AjaxMixin, View):
+    """
+    Returns the user id in the session.
+    """
+
+    def get(self, request, *args, **kwargs):
+        return self.ajax_response()
+
+    def get_ajax_context(self, **context):
+        return super().get_ajax_context(
+            user_id=self.request.user.pk,
```

Comment:
No, I use the value 'null' in angular to work with it. So I can set the current user to the return value of this method and do not have to check for different status codes.

If the view is requested by an anonymous user self.request.user is the AnonymousUser which pk is none